### ISSUE TITLE
Include order in C++

### ISSUE THREAD

**Issue by flaviojs** [ja2-stracciatella/ja2-stracciatella#939]
In rust I'm grouping as std, then other crates, then the crate. Inside each group it's sorted alphabetically.
Main reasons:
1) easier to know where to find docs and source
2) it's intuitive:
2.1) code comes after includes (use) so it also shows the closeness to your code
2.2) alphabetical order is means you don't have to read all the includes in the group

What about in C++?
CONTRIBUTING.md says "Sort includes by type (project, system) and alphabetically."
Is this supposed to be like rust but with the group order reversed?
That is: our includes, dependency includes, core includes

**Comment by lynxlynxlynx** [comment]
Yes. My usual order is: same dir, other dirs, system, so paths in includes aren't scattered all around due to sorting. In some projects the counterpart header has a separate place too, since it makes sense for it to be prominent.

In the existing code it's a mess, with system includes sometimes even in the middle of local ones — nobody cleaned up the style. If you want to unify it, go ahead. I for one don't care if the supraordering goes one way or the other.

**Comment by flaviojs** [comment]
I'll fix the order of the include sections I modify, which right now is C/C++ includes.
